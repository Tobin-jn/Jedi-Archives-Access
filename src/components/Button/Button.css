@font-face {
  font-family: "Galactic Basic";
  src: url("./images/galactic_basic/galbasic.ttf") format("ttf"),
}

.nav-button {
  position: relative;
  margin: 0 .2rem;
  width: 82px;
  height: 20px;
  color: #ADD8E6;
  border-radius: 5px;
  overflow: hidden;
  transition: .5s;
  text-decoration: none;
  padding: 0;
}

.nav-button-text {
  font-family: Andale Mono,AndaleMono,monospace;
  margin: 0 5px 0 0;
  text-align: center;
  text-decoration: none;
  font-size: .8rem;
  transition: .5s;
  text-transform: uppercase;
  text-align: center;
  line-height: 20px;
}

.nav-button:hover {
  cursor: pointer;
}

.nav-button img {
  width: 50px;
  height: 50px;
}

.button-selected {
  border: 1px solid rgba(173, 216, 230, .5);
  box-shadow: 0 10px 5px #ADD8E6;
}

.nav-button-text:before {
  content:attr(data-text);
  font-family: 'Galactic Basic';
  font-size: .3rem;
  color: #ADD8E6;
  position: absolute;
  top: 0px;
  left: 6px;
  opacity: 0;
  transition: .5s;
}

.nav-button-text:hover {
  color: transparent;
}

.nav-button-text:hover:before {
  opacity: 1;
}


