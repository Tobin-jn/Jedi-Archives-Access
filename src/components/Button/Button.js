import React from 'react';
import './Button.css';
import PropTypes from 'prop-types';

const Button = ({handleSelection, buttonName, currentSelection}) => {

  let buttonState;
  let buttonPath = `./${buttonName}.png`;

  if (currentSelection === buttonName) {
    buttonState = 'button-selected'
  } else {
    buttonState = ''
  }
  
  return(
    <div 
      className={`nav-button ${buttonState}`}
      onClick={() => {
        handleSelection(buttonName)
      }}
    >
      <h3 
        data-text={buttonName}
        className='nav-button-text'>{buttonName}
      </h3>
    </div>
  );
}

Button.propTypes = {
  buttonName: PropTypes.string.isRequired,
  handleSelection: PropTypes.func.isRequired,
  currentSelection: PropTypes.string.isRequired
};

export default Button;