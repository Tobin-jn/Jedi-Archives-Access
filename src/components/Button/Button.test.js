import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  let wrapper;
  let mockHandleSelection;

  beforeEach(() => {
    mockHandleSelection = jest.fn()

    wrapper = shallow(<Button 
      buttonName="people"
      handleSelection={mockHandleSelection}
      currentSelection="people"
    />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a className of button-selected if selected', () => {
    expect(wrapper.hasClass('nav-button button-selected')).toEqual(true);
  })

  it('should have a className of nav-button if not selected', () => {
    wrapper = shallow(<Button 
      buttonName="people"
      handleSelection={mockHandleSelection}
      currentSelection="planets"
    />)
    expect(wrapper.hasClass('nav-button')).toEqual(true);
  })

  it('should render a button', () => {
    expect(wrapper.find('div').length).toEqual(1);
  })

  it('should call handleSelection on click', () => {
    wrapper.find('.nav-button').simulate('click');
    expect(mockHandleSelection).toHaveBeenCalled();
  })
})
