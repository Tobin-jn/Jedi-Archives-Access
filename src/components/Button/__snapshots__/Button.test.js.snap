// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<div
  aria-label="navigation-button"
  className="nav-button button-selected"
  onClick={[Function]}
>
  <h3
    className="nav-button-text"
    data-text="people"
  >
    people
  </h3>
</div>
`;
